journal/src/test/java/io/camunda/zeebe/journal/file/SegmentsManagerTest.java

```diff
@@ -0,0 +1,160 @@
+    // scenario.
+    createSegmentsManager(0).open();
+
+    // then
+    final File logDirectory = directory.resolve("data").toFile();
+    assertThat(logDirectory)
+        .isDirectoryNotContaining(
+            file -> JournalSegmentFile.isDeletedSegmentFile(JOURNAL_NAME, file.getName()))
+        .isDirectoryContaining(
+            file -> JournalSegmentFile.isSegmentFile(JOURNAL_NAME, file.getName()));
+  }
+
+  @Test
+  void shouldDetectCorruptionAtDescriptorWithAckedEntries() throws Exception {
+    // given
+    final var journal = openJournal(1);
+    final long index = journal.append(data).index();
+
+    journal.close();
+    final File dataFile = directory.resolve("data").toFile();
+    final File logFile =
+        Objects.requireNonNull(dataFile.listFiles(f -> f.getName().endsWith(".log")))[0];
+    LogCorrupter.corruptDescriptor(logFile);
+
+    // when/then
+    final var segments = createSegmentsManager(index);
+    assertThatThrownBy(segments::open).isInstanceOf(CorruptedLogException.class);
+  }
+
+  @Test
+  void shouldNotThrowExceptionWhenCorruptionAtNotAckEntries() throws Exception {
+    // given
+    final var journal = openJournal(1);
+    final var index = journal.append(data).index();
+    journal.append(data);
+
+    journal.close();
+    final File dataFile = directory.resolve("data").toFile();
+    final File logFile =
+        Objects.requireNonNull(dataFile.listFiles(f -> f.getName().endsWith("2.log")))[0];
+    LogCorrupter.corruptDescriptor(logFile);
+
+    // when/then
+    final var segments = createSegmentsManager(index);
+    assertThatNoException().isThrownBy(segments::open);
+    assertThat(segments.getFirstSegment().index()).isEqualTo(index);
+    assertThat(segments.getLastSegment().index()).isEqualTo(index);
+  }
+
+  @Test
+  void shouldNotThrowExceptionWhenCorruptionAtDescriptorWithoutAckedEntries() throws Exception {
+    // given
+    final var journal = openJournal(1);
+    journal.close();
+    final File dataFile = directory.resolve("data").toFile();
+    final File logFile =
+        Objects.requireNonNull(dataFile.listFiles(f -> f.getName().endsWith(".log")))[0];
+    LogCorrupter.corruptDescriptor(logFile);
+
+    // when/then
+    final var segments = createSegmentsManager(0);
+    assertThatNoException().isThrownBy(segments::open);
+    assertThat(segments.getFirstSegment().index()).isEqualTo(1);
+    assertThat(segments.getFirstSegment().lastIndex()).isEqualTo(0);
+  }
+
+  @Test
+  void shouldHandlePartiallyWrittenDescriptor() throws Exception {
+    // given
+    final File dataFile = directory.resolve("data").toFile();
+    assertThat(dataFile.mkdirs()).isTrue();
+    final File emptyLog = new File(dataFile, "journal-1.log");
+    assertThat(emptyLog.createNewFile()).isTrue();
+
+    // when
+    final var segments = createSegmentsManager(0);
+
+    // then
+    assertThatNoException().isThrownBy(segments::open);
```

Comment:
:+1: Uh, I didn't know this one! I only knew the `assertThatCode(() -> {}).doesNotThrowException()`. TIL